account_invoice_secondary_unit/models/sale_order_line.py

```diff
@@ -12,6 +13,16 @@ def _prepare_invoice_line(self, qty):
         self.ensure_one()
         res = super(SaleOrderLine, self)._prepare_invoice_line(qty)
         if self.secondary_uom_id:
-            res['secondary_uom_qty'] = self.secondary_uom_qty
+            res['secondary_uom_qty'] = self._get_secondary_uom_qty()
             res['secondary_uom_id'] = self.secondary_uom_id.id
         return res
+
+    @api.multi
+    def _get_secondary_uom_qty(self):
```

Comment:
This is practically not needed, but to be clear about the usage/intention of the method, we may add `self.ensure_one()` since this method doesn't expect to handle multiple records.